service/product/service.go

```diff
@@ -56,3 +56,109 @@ func CreateProduct(w http.ResponseWriter, r *http.Request) {
 	log.Printf("Product %s created successfully", p.Name)
 	helper.BaseResponse(w, response, http.StatusOK)
 }
+
+func DeleteProduct(w http.ResponseWriter, r *http.Request) {
+	var p viewmodels.Product
+	err := json.NewDecoder(r.Body).Decode(&p)
+	if err != nil {
+		log.Println("Failed to decode request body: ", err)
+		response, _ := json.Marshal(viewmodels.BaseResponse{
+			StatusCode: statusCodes.FailedToDecodeRequestBody,
+			Message:    statusCodes.StatusCodes[statusCodes.FailedToDecodeRequestBody],
+		})
+		helper.BaseResponse(w, response, http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	_ = *helper.CheckToken(&w, r)
```

Comment:
what this function do?

Reply:
Authorizes token. It returns claims, in some functions I use those claims, in other, where not needed, it just checks. It handles errors and response inside.